plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/AbstractTestIcebergConnectorTest.java

```diff
@@ -1822,6 +1822,120 @@ public void testIncorrectIcebergFileSizes()
         dropTable("test_iceberg_file_size");
     }
 
+    @Test
+    public void testJoinWithPartitionTable()
```

Comment:
This test case is testing the changes made but indirectly and it'll be difficult to identify in future what this is supposed to exactly test. This join test uncovers the problem because it tries to read table statistics which is where the error happens.

I propose instead directly testing that `SHOW STATS` returns correct results for all data-types and possible transforms.
Take a look at existing test methods names `test*Statistics` in this class for how you can structure this.

cc: @findepi This also ties into having dedicated tests for table statistics similar to the TestHiveTableStatistics product test.

Reply:
> I propose instead directly testing that SHOW STATS returns correct results for all data-types and possible transforms.

👍

Reply:
@hashhar Thanks advices and patient, i will look at the existing test methods